src/bin/electrs.rs

```diff
@@ -62,12 +62,28 @@ fn run_server(config: &Config) -> Result<()> {
     let query = Query::new(app.clone(), &metrics, tx_cache, config.txid_limit);
 
     let mut server = None; // Electrum RPC server
+
+    // If we see this more new blocks than this, don't look for scripthash
+    // changes. Just have clients reconnect.
+    //
+    // This many header changes means we're either not fully synced, or there
+    // has been abnormally large reorg of the blockchain.
+    let MAX_SCRIPTHASH_BLOCKS = 10;
+
     loop {
-        app.update(&signal)?;
-        query.update_mempool()?;
-        server
-            .get_or_insert_with(|| RPC::start(config.electrum_rpc_addr, query.clone(), &metrics))
-            .notify(); // update subscribed clients
+        let (headers_changed, new_tip) = app.update(&signal)?;
+        let txs_changed = query.update_mempool()?;
+        let rpc = server
+            .get_or_insert_with(|| RPC::start(config.electrum_rpc_addr, query.clone(), &metrics));
+        if headers_changed.len() > MAX_SCRIPTHASH_BLOCKS {
```

Comment:
Is the server being open before Electrum reaches the tip of its daemon?
Id so, then isn't this check redundant? (clients will be disconnected anyway. And if there's a re-org, why not sending a notification?)

Reply:
> Is the server being open before Electrum reaches the tip of its daemon?

Yes, with my testing clients can connect before we're synced up. It takes a very long time to sync the final blocks if we produce notifications at the same time (in the normal case it doesn't take long, because we've already seen and cached the transactions when they entered mempool).

Also there is also no way to know from here if any clients are connected, so this was the simplest solution. 

As for reorg, we can pretty much assume that a >= 10 block won't happen on mainnet, and if we disconnect the clients on such a big reorg its no big deal. We can also increase `MAX_SCRIPTHASH_BLOCKS` if its too low.